Sources/CoreMetrics/Metrics.swift

```diff
@@ -173,7 +173,7 @@ public class Gauge: Recorder {
 }
 
 public enum TimeUnit {
-    case nanoseconds, milliseconds, seconds, minutes, hours, days
+    case nanoseconds, microseconds, milliseconds, seconds, minutes, hours, days
```

Comment:
this breaks public API. It's an odd oversight that this is missing but we can't just add this here. CC @tomerd / @ktoso what do we do?

Reply:
hey @weissi I was talking g to @jckarter the other day on something else and I believe I heard him say that adding an enum case is considered an additive change. the api-digester also didn't complain. wdyt?

Reply:
@tomerd it's definitely not. In the Swift dialect that they use in the OSes it is (unless the `enum` is `@frozen`). In the version we use with SwiftPM however adding an enum case is  breaking. Imagine a user wrote this code

```
            switch displayUnit {
            case .days: print("days")
            case .hours: print("hours")
             case .minutes: print("minutes")
             case .seconds: print("seconds")
             case .milliseconds: print("milliseconds")
             case .nanoseconds: print("nanoseconds")
             }
```

this previously compiled and won't after this patch :(

Reply:
Right, adding a case is only resilient if you have library evolution enabled for your module. Maybe we could add an option to opt in to "unfrozen" behavior in general.

Reply:
@jckarter I think that’d be awesome. However in this case we just messed up. This should’ve never been an enum. Should’ve been a struct with static lets and a hidden enum :(.

Reply:
ugh painful. @cburrows how soon do you need this?

Reply:
@weissi really interesting the api-digester did not complain https://ci.swiftserver.group/job/swift-metrics-api-breakage-prb/16/console will need to dig into why

Reply:
Yep. So... I don't think we have much choice other than releasing this as an 2.0.

Reply:
@tomerd AIUI, the API digester digests the API according to the `-enable-library-evolution` rules, so it may not be entirely reliable for packages in its current state. Is that right @nkcsgexi?

Reply:
@tomerd that's a pretty interesting bug, I'll have a look.

@ktoso / @tomerd yes, I'd also vote for going `2.0.0` ASAP. Right now, `swift-metrics` doesn't have that many dependents (search `swift-metrics` on http://swift-dependency-graph.opticalaberration.com , then tap on the box to invert from dependencies to dependents). Mostly:

- Vapor 4 ecosystem: still in development but tagging 4.0.0 soon
- Smoke framework 2: still in development
- swift-redistack

Vapor 4 and Smoke 2 could just pick up `from: "2.0.0"` and swift-redistack could do a `1.0.0 ..< 3.0.0` which includes all 1.x.y and all 2.x.y versions. swift-redistack just needs to be careful not to switch on this `enum` and apart from that 1.x and 2.x are totally compatible so in reality I don't think it'll be a problem. Also: We can talk to all those maintainers straight away so I don't think there's be a big chance of splitting the ecosystem with many dependencies only accepting 1.x releases.

Reply:
Apologies, I missed a few dependents on the first scan:
- swiftbox
- statsd-client
- swiftprometheus

but again, I'm pretty sure we can get everybody to accept `1.0.0 ..< 3.0.0` and make sure they work with both 1.x and 2.x if they can't outright switch to `from: "2.0.0"`

Reply:
For SwiftPrometheus we can switch to 2.0.0. And I think with Vapor it’s the same.

Reply:
cc @tanner0101 @tachyonics @Mordil

Reply:
I'm not relying on `TimeUnit` directly in RediStack, so this won't be a problem. RediStack also hasn't had a 1.0.0 so I can do it immediately.

We'll want to write an issue to update this for 3.0, though (or solve it right now for 2.0)

Reply:
If we're touching this now anyways, I'd propose to move to a strategy that allows us to not break public API in the future, and not wait with that until a 3.x release.

Reply:
This change won't break the Smoke* repositories; we have only just started depending on SwiftMetrics and we are close to but not at stabilization stage for our 2.0 versions.

Reply:
That's not a breaking change for Swiftbox, it does not rely dorectly on TimeUnit enum. Thanks for letting me know.

Reply:
ping @tanner0101

Reply:
@tomerd this should be fine. Vapor 4 exports Metrics so upgrading to 2 is technically breaking but we're still in RC phase and I doubt any Vapor users are switching on this enum.

Reply:
@jckarter it would be a game changer if SPM libraries could opt in to the "unfrozen" behavior that resilient libraries can. Vapor/* packages have to use tons of enum-like structs in order to work around this. They are quite verbose and hurt code readability where they are both defined and used. In places where we forget to do this and use public enums we get put in difficult situations like this one.